.github/workflows/android-build.yml

```diff
@@ -41,20 +45,24 @@ jobs:
       - name: Check out Git repository
         uses: actions/checkout@v4
 
+      - name: Install dependencies in root
+        run: yarn install --immutable
+
       - name: Setup Java 17
         uses: actions/setup-java@v3
         with:
-          distribution: 'zulu'
+          distribution: "zulu"
           java-version: 17
 
       - name: Restore Reanimated node_modules from cache
         uses: actions/cache@v3
         with:
-          path: node_modules
-          key: ${{ runner.os }}-android-node-modules-reanimated-${{ hashFiles('yarn.lock') }}
+          path: packages/react-native-reanimated/node_modules
```

Comment:
cc @tomekzaw

If you could explain to me (preferably offline) how does the memoization work here.